File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/validation/custom/DurationPattern.kt
package io.micronaut.docs.validation.custom

// tag::imports[]
import jakarta.validation.Constraint
import kotlin.annotation.AnnotationRetention.RUNTIME
// end::imports[]

// tag::class[]
@Retention(RUNTIME)
@Constraint(validatedBy = []) // <1>
annotation class DurationPattern(
    val message: String = "invalid duration ({validatedValue})" // <2>
)
// end::class[]

File: test-suite-python/src/test/python/micronaut/docs/validation/custom/DurationPattern.py
# tag::imports[]
from jakarta.validation import Constraint

from .MyValidatorBean import MyValidatorBean
# end::imports[]


# tag::class[]
@Constraint(validatedBy=[MyValidatorBean])  # <1>
def DurationPattern(message: str = "invalid duration ({validatedValue})"):  # <2>
    def decorator(target):
        return target

    return decorator
# end::class[]
